# based on a patch from the following location:
http://www.mail-archive.com/u-boot@lists.denx.de/msg17570.html

[U-Boot] [PATCH] nand: fixed failed reads on corrected ECC errors in nand_util.c

Valeriy Glushkov
Tue, 14 Jul 2009 03:50:23 -0700

Signed-off-by: Valeriy Glushkov <g...@lstec.com>
Signed-off-by: Paulraj, Sandeep <s-paul...@ti.com>
---
 drivers/mtd/nand/nand_util.c |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/mtd/nand/nand_util.c b/drivers/mtd/nand/nand_util.c
index fc16282..694ead6 100644
--- a/drivers/mtd/nand/nand_util.c
+++ b/drivers/mtd/nand/nand_util.c
@@ -564,10 +564,10 @@ int nand_read_skip_bad(nand_info_t *nand
 
 	if (len_incl_bad == *length) {
 		rval = nand_read (nand, offset, length, buffer);
-		if (rval != 0)
-			printf ("NAND read from offset %zx failed %d\n",
-				offset, rval);
-
+		if (!rval || rval == -EUCLEAN)
+			return 0;
+		printf ("NAND read from offset %llx failed %d\n",
+			offset, rval);
 		return rval;
 	}
 
@@ -588,7 +588,7 @@ int nand_read_skip_bad(nand_info_t *nand
 			read_length = nand->erasesize - block_offset;
 
 		rval = nand_read (nand, offset, &read_length, p_buffer);
-		if (rval != 0) {
+		if (rval && rval != -EUCLEAN) {
 			printf ("NAND read from offset %zx failed %d\n",
 				offset, rval);
 			*length -= left_to_read;
